lib/Backend/SccLiveness.cpp

```diff
@@ -781,30 +781,36 @@ SCCLiveness::FoldIndir(IR::Instr *instr, IR::Opnd *opnd)
         {
             return false;
         }
-        indir->SetOffset(offset);
+        int32 newOffset = 0;
+        if(!Int32Math::Add(offset,indir->GetOffset(), &newOffset))
```

Comment:
I don't understand this part.
The check above already does add the indir offset and the scaled index.
This added check returns false if an additional add doesn't overflow... And if it does overflow, use it ?